Core/include/Managers/ComponentManager.h

```diff
@@ -1,24 +1,90 @@
 #pragma once
```

Comment:
This is repurposing of the ComponentManager file, naming can be changed.
The point is to extract the code that adds and setup components to an external function. While our current components are fairly simple without many parameters, that doesn't mean future components won't get complex.

The Light component right now is fairly complex with 8 parameters that can be changed. Do we want to potentially have to modify up to 8 lines of code every time we add/change a light component? No we want to add/change one line for that component.

Yes this does mean we will have to pass in the same data that had been defined outside the function call, but that's the trade off for extracting these into their own functions,

Reply:
What I meant by my first paragraph is that I dont think this class is necessary at all, we're essentially doing the same thing in a different place.

And to your example for the light I dont think thats the case, if i'm reading into this file's current implementation correctly we'd just be editing the 8 lines somewhere else, hence why I thought we're just adding complexity for barely any gain

Reply:
The current implementation of that function, tbh this file in general, is not in scope of this PR. Previously they were going to be left as TODO's. Having gone through I recognize there is definitely a lot of boilerplating this way, but it's a function library so that's fine.

AddLightComponent() will be designed to handle light params being passed into it, AddCameraComponent() as well.